src/environment/browser-router.js

```diff
@@ -27,7 +27,7 @@ export default (
   } = history.location;
 
   // Strip the basename from the initial pathname
-  const pathname = basename ? fullPathname.replace(basename, '') : fullPathname;
+  const pathname = basename ? fullPathname.replace(new RegExp(`^${basename}`), '') : fullPathname
```

Comment:
I think we can simplify this to avoid dynamically allocating a `RegExp` instance by doing a substring match + replace instead.

```js
const pathname = basename && fullPathname.indexOf(basename) === 0
  ? fullPathname.slice(basename.length)
  : fullPathname;

Reply:
Smart! I've pushed the changes you requested.